### PR TITLE
Better image tags

### PR DESCRIPTION
**Pull request by mattiecnvr**
### Description

Changes image tags to look like this
`9b61e86-20230202-174933`
Where the first part is the short sha, middle is the year-month-day, and last part is the timestamp (hours mins secs) (to help distinguish which image came first when there are multiple on the same day in use).

### CODE REVIEW

**Review by asaj — COMMENTED** [review]
Thanks for doing this!

File: .github/workflows/monorepo-docker.yml [hunk 1]
```diff
@@ -52,7 +56,8 @@ jobs:
           tags: |
             type=ref,event=branch
             type=ref,event=pr
-            type=sha
+            type=schedule,pattern={{date 'YYYYMMDD-HHmmss'}}
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply .github/workflows/monorepo-docker.yml]: Forgot to remove that one. Fixed.

**Comment by mattiecnvr** [comment]
What the docker command currently looks like in the pipeline

```
/usr/bin/docker buildx build --cache-from type=gha --cache-to type=gha,mode=max --file ./Dockerfile --iidfile /tmp/docker-build-push-mbkhlP/iidfile --label org.opencontainers.image.title=hyperlane-monorepo --label org.opencontainers.image.description=The home for Hyperlane core contracts, sdk packages, and other infrastructure --label org.opencontainers.image.url=https://github.com/hyperlane-xyz/hyperlane-monorepo --label org.opencontainers.image.source=https://github.com/hyperlane-xyz/hyperlane-monorepo --label org.opencontainers.image.version=pr-1733 --label org.opencontainers.image.created=2023-02-02T19:40:14.319Z --label org.opencontainers.image.revision=ae66ce26d59118a252345f296de88bfaf71ba76c --label org.opencontainers.image.licenses=NOASSERTION --provenance false --tag gcr.io/abacus-labs-dev/hyperlane-monorepo:pr-1733 --tag gcr.io/abacus-labs-dev/hyperlane-monorepo:ae66ce2-20230202-194013 --metadata-file /tmp/docker-build-push-mbkhlP/metadata-file --push ./
```

**Review by asaj — APPROVED** [review]
Nice!

**Pull request merged**